src/main/java/org/example/Contact.java

```diff
@@ -0,0 +1,62 @@
+package org.example;
+
+
+import java.time.LocalDateTime;
+
+public abstract class Contact {
+
+    public static final String EMPTY_NUMBER = "[no number]";
+    private String phoneNumber;
+    private final LocalDateTime creationDate;
+    private LocalDateTime lastEditDate;
+
+    public Contact(String phoneNumber) {
+        setPhoneNumber(phoneNumber);
+        creationDate = LocalDateTime.now();
+        lastEditDate = creationDate;
+    }
+
+    public String getPhoneNumber() {
+        return phoneNumber;
+    }
+
+    public void setPhoneNumber(String phoneNumber) {
+        if (isValidPhoneNumber(phoneNumber)) {
+            this.phoneNumber = phoneNumber;
+        } else {
+            this.phoneNumber = EMPTY_NUMBER;
+        }
+    }
+
+//    Rules of the regex below
```

Comment:
do we really need these comments? Method name should let us know clearly what we are doing

Reply:
I just added these comments to be in sight, because I was struggling with this regex))